docs/code-standards-policy.rst

```diff
@@ -118,36 +177,308 @@ using the debops.apt_install_ role MUST be defined as (or similar):
+on a provider role, MUST additionally contain the word ``dependent`` in their
+variable name. E.g.:
+
+.. code-block:: yaml
+
+   apt_preferences__dependent_list: []
+
+Variables which are meant to define dependent variables in the consumer role
+are named after the dependent variable of the provider role prefixed with the
+consume role name. E.g.
 
 .. code-block:: yaml
 
-   application__apt_preferences__dependent_list:
+   nginx__apt_preferences__dependent_list:
      - package: 'nginx nginx-*'
        backports: [ 'wheezy', 'precise' ]
 
-Then, in your role's playbook, you can add the debops.apt_preferences_ role as
-a dependency and pass a specific configuration to it:
+
+.. _debops_policy__ref_code_standards_default_variable_documentation:
+
+Variable documentation
+~~~~~~~~~~~~~~~~~~~~~~
+
+For each role the `DebOps Documentation`_ is will include a page which
+documents the default variables. This page is generated from the role's
+:file:`defaults/main.yml` file with help of yaml2rst_. The entire comment of
+the defauls file is thereby interpreted as reStructuredText_ and then rendered
+via Sphinx_.
+
+Each variable comment is started with a ``.. envvar::`` reference anchor
+followed by the name of the variable. This construct allows any documentation
+page to reference the variable via ``:envvar:`varname``` which :program:`Sphinx`
+will translate into a link pointing to the variable description. Below the
+anchor a comment should describe the purpose of the variable, accepted values,
+side effects and so on. Within the comment all reStructuredText constructs
+supported by :program:`Sphinx` can be used. An example variable definition would
+eventually look like this:
+
+.. code-block:: yaml
+
+    # .. envvar:: ferm__flush [[[
+    #
+    # Should ferm-rules be flushed when :program:`ferm` is disabled? The default is true,
+    # but you may need set both :envvar:`ferm__enabled` and this to ``False`` if you are
+    # running in some container and are not allowed to change :command:`iptables`.
+    ferm__flush: '{{ ferm__enabled | bool }}'
+                                                                   # ]]]
+
+Related default variables should be grouped to sections for a better overview
+and easier navigation. For example it makes sense to distinguish packaging and
+network related variables. Of course additional or different section titles
+might be meaningful for the individual role:
+
+.. code-block:: yaml
+
+    # APT packages [[[
+    # ----------------
+
+    [... packaging related variables ...]
+
+                                                                   # ]]]
+    # Network configuration [[[
+    # -------------------------
+
+    [... networking related variables ...]
+                                                                   # ]]]
+
+.. _debops_policy__ref_code_standards_dependent_variables:
+
+Dependent variables
+-------------------
```

Comment:
Is it only me or someone else also thinks this is a strange expression? Does maybe someone have a better idea?